src/number_theory/legendre_symbol/quadratic_reciprocity.lean

```diff
@@ -32,32 +29,29 @@ Also proven are conditions for `-1` and `2` to be a square modulo a prime,
 The proof of quadratic reciprocity implemented uses Gauss' lemma and Eisenstein's lemma
 -/
 
-open function finset nat finite_field zmod
-open_locale big_operators nat
+open finset nat char
 
 namespace zmod
 
+/-- We have that `ring_char (zmod n) = n`. -/
+lemma ring_char_zmod_n (n : ℕ) : ring_char (zmod n) = n :=
+by { rw ring_char.eq_iff, exact zmod.char_p n, }
+
 variables (p q : ℕ) [fact p.prime] [fact q.prime]
 
 /-- Euler's Criterion: A unit `x` of `zmod p` is a square if and only if `x ^ (p / 2) = 1`. -/
 lemma euler_criterion_units (x : (zmod p)ˣ) :
   (∃ y : (zmod p)ˣ, y ^ 2 = x) ↔ x ^ (p / 2) = 1 :=
 begin
-  cases nat.prime.eq_two_or_odd (fact.out p.prime) with hp2 hp_odd,
-  { substI p, refine iff_of_true ⟨1, _⟩ _; apply subsingleton.elim },
-  obtain ⟨g, hg⟩ := is_cyclic.exists_generator (zmod p)ˣ,
-  obtain ⟨n, hn⟩ : x ∈ submonoid.powers g, { rw mem_powers_iff_mem_zpowers, apply hg },
-  split,
-  { rintro ⟨y, rfl⟩, rw [← pow_mul, two_mul_odd_div_two hp_odd, units_pow_card_sub_one_eq_one], },
-  { subst x, assume h,
-    have key : 2 * (p / 2) ∣ n * (p / 2),
-    { rw [← pow_mul] at h,
-      rw [two_mul_odd_div_two hp_odd, ← card_units, ← order_of_eq_card_of_forall_mem_zpowers hg],
-      apply order_of_dvd_of_pow_eq_one h },
-    have : 0 < p / 2 := nat.div_pos (fact.out (1 < p)) dec_trivial,
-    obtain ⟨m, rfl⟩ := dvd_of_mul_dvd_mul_right this key,
-    refine ⟨g ^ m, _⟩,
-    rw [mul_comm, pow_mul], },
+  by_cases hc : p = 2,
+  { substI hc,
+    simp only [eq_iff_true_of_subsingleton, exists_const], },
+  { have h₀ := finite_field.unit_is_sqare_iff (by rwa ring_char_zmod_n) x,
```

Comment:
Can you fix this typo in the lemma name?

```suggestion
  { have h₀ := finite_field.unit_is_square_iff (by rwa ring_char_zmod_n) x,
```

Reply:
Thanks for pointing this out. Fixed it.